### PR TITLE
added packages

### PR DESCRIPTION
**Pull request by kzidane**
* composer
* hub
* node v7.6.0
* virtualenv
* pyenv and python 3.6

### CODE REVIEW

**Comment by kzidane** [comment]
Not ready yet. Will integrate `pyenv` and `rbenv` shortly after testing...

**Comment by kzidane** [comment]
@dmalan this should be ready, unless you have any takes?

File: files/etc/profile.d/ide50.sh [hunk 1]
```diff
@@ -11,7 +11,7 @@ if [ "$(id -u)" != "0" ]; then
     # set umask
     umask 0077
 
-    export PATH=/opt/cs50/bin:$PATH:$HOME/.local/bin ;;
+    export PATH=/opt/cs50/bin:$PATH:$HOME/.local/bin
```
Comment (dmalan) [inline files/etc/profile.d/ide50.sh]: Best to quote env vars in case they have spaces.

File: files/etc/profile.d/ide50.sh [hunk 2]
```diff
@@ -53,7 +53,13 @@ alias pip=pip3
...
 alias python=python3
 export PYTHONDONTWRITEBYTECODE=1
-export PYTHONPATH="/home/ubuntu/.local/lib/python3.4/site-packages:/usr/lib/python3/dist-packages:/usr/local/lib/python3.4/dist-packages"
+
+# unset so user doesn't have to restart IDE
+unset PYTHONPATH
+
+# pyenv
+export PYENV_ROOT=/opt/pyenv
+export PATH=$PYENV_ROOT/bin:$PYENV_ROOT/shims:$PATH
```
Comment (dmalan) [inline files/etc/profile.d/ide50.sh]: quotes

File: update50 [hunk 3]
```diff
@@ -52,6 +52,60 @@ sudo apt-get update
...
+        libssl-dev \
+        llvm \
+        wget \
+        xz-utils \
+        zlib1g-dev
+    wget -P /tmp https://github.com/yyuu/pyenv/archive/master.zip
+    unzip -q -d /tmp /tmp/master.zip
+    rm -f /tmp/master.zip
+    sudo mv /tmp/pyenv-master /opt/pyenv
+    sudo chmod a+x /opt/pyenv/bin/pyenv
```
Comment (dmalan) [inline update50]: Is a `chmod a+rX` also necessary potentially? (Not sure of umask.)

File: update50 [hunk 4]
```diff
@@ -60,9 +114,9 @@ echo "Installing Python packages..."
 # clean up pip's build cache
 /usr/bin/sudo rm -rf /tmp/pip_build_root/*
 
-# upgrade pip3
-/usr/bin/sudo easy_install-3.4 -U pip
-
+# use pip for python3.6
+unset PYTHONPATH
+export PATH=$PYENV_ROOT/bin:$PYENV_ROOT/shims:$PATH
```
Comment (dmalan) [inline update50]: quotes

File: update50 [hunk 5]
```diff
@@ -60,9 +114,9 @@ echo "Installing Python packages..."
 # clean up pip's build cache
 /usr/bin/sudo rm -rf /tmp/pip_build_root/*
 
-# upgrade pip3
-/usr/bin/sudo easy_install-3.4 -U pip
-
+# use pip for python3.6
+unset PYTHONPATH
```
Comment (dmalan) [inline update50]: Why unsetting again?

Comment (kzidane) [inline update50]: We're now setting Python 3.6 to be the default. Keeping `PYTHONPATH` set to [paths for other Python versions](https://github.com/cs50/ide50/blob/f77e01331e1ea73f01259604e3b4bfe74511b3a7/files/etc/profile.d/ide50.sh#L56) confuses it, causing an error.

File: update50 (hunk 3 above)
Comment (kzidane) [inline update50]: `root`'s umask is `0022`, so we should be good even without that `chmod`, especially that `/opt/pyenv/bin/pyenv` is actually a symlink.

**Comment by kzidane** [comment]
All env var values should be quoted now!

**Pull request merged**